### PR TITLE
Remove GAed feature gate CronJobControllerV2

### PR DESCRIPTION
**Pull request by pacoxu**
/kind cleanup
/sig apps
/assign @alaypatel07 @soltysh

follow up of  #102529

```release-note
NONE
```

```
- [KEP]: https://github.com/kubernetes/enhancements/tree/master/keps/sig-apps/19-Graduate-CronJob-to-Stable 
```

### CODE REVIEW

**Review by alaypatel07 — COMMENTED** [review]
thanks for taking this @pacoxu 

/lgtm

**Comment by pacoxu** [comment]
Rebased for conflicts.

**Comment by pacoxu** [comment]
/priority important-soon
since it should be done in 1.23

**Review by alaypatel07 — COMMENTED** [review]
/lgtm

File: cmd/kube-controller-manager/app/batch.go [hunk 1]
```diff
@@ -41,23 +39,13 @@ func startJobController(ctx context.Context, controllerContext ControllerContext
...
-	cjc, err := cronjob.NewController(
+	cj2c, err := cronjob.NewControllerV2(controllerContext.InformerFactory.Batch().V1().Jobs(),
+		controllerContext.InformerFactory.Batch().V1().CronJobs(),
 		controllerContext.ClientBuilder.ClientOrDie("cronjob-controller"),
 	)
 	if err != nil {
-		return nil, true, fmt.Errorf("error creating CronJob controller: %v", err)
+		return nil, true, fmt.Errorf("error creating CronJob controller V2: %v", err)
 	}
-	go cjc.Run(ctx.Done())
```
Comment (pacoxu) [inline cmd/kube-controller-manager/app/batch.go:61]: Remove line 55-61
Remove the `if` in line 44.

**Review by soltysh — APPROVED** [review]
/lgtm
/approve

**Pull request merged**